test/unit/custom_spec.js

```diff
@@ -0,0 +1,99 @@
+import { buildGetDocumentParams } from './test_utils';
+import { DOMCanvasFactory } from '../../src/display/dom_utils';
+import { getDocument } from '../../src/display/api';
+import { isNodeJS } from '../../src/shared/util';
+import { PDFJS } from '../../src/display/global';
+
+function getTopLeftPixel(canvasContext) {
+  var imgData = canvasContext.getImageData(0, 0, 1, 1);
+  var r = imgData.data[0];
+  var g = imgData.data[1];
+  var b = imgData.data[2];
+  var a = imgData.data[3];
+  return { r, g, b, a};
+}
+
+describe('custom canvas rendering', function() {
+  var transparentFileName = 'transparent.pdf';
+  var transparentGetDocumentParams = buildGetDocumentParams(transparentFileName);
+
+  var CanvasFactory;
+  var loadingTask;
+  var pdfDocument, page;
+
+  beforeAll(function(done) {
+    if (isNodeJS()) {
+      PDFJS.pdfjsNext = true;
+      // NOTE: To support running the canvas-related tests in Node.js,
+      // a `NodeCanvasFactory` would need to be added (in test_utils.js).
+    } else {
+      CanvasFactory = new DOMCanvasFactory();
+    }
+    loadingTask = getDocument(transparentGetDocumentParams);
+    loadingTask.promise.then(function(doc) {
+      pdfDocument = doc;
+      return pdfDocument.getPage(1);
+    }).then(function(data) {
+      page = data;
+      done();
+    }).catch(function (reason) {
+      done.fail(reason);
+    });
+  });
+
+  afterAll(function(done) {
+    CanvasFactory = null;
+    done();
```

Comment:
Please clean up the `loadingTask` properly here, i.e. `loadingTask.destroy().then(done):`.